### PR TITLE
Do not process too fresh buildrequests

### PR DESCRIPTION
**Pull request by rail**

### CODE REVIEW

File: bbb/services.py [hunk 1]
```diff
@@ -285,6 +285,16 @@ def reflectTasks(self):
             # If takenUntil isn't set, this task has either never been claimed
             # or got cancelled.
             if not t.takenUntil:
+                # If the buildrequest is showing complete, there is a
+                # possibility, that the build was completed before takenUntil
+                # was updated by BBListener. To avoid this we can try to avoid
+                # processing the buildrequest for 5 minutes.
+                if arrow.now() < arrow.get(t.processedDate).replace(minutes=5):
+                    log.debug(
```
Comment (lundjordan) [inline bbb/services.py]: this probably won't happen often. I wonder if this should be an info/warn so we see it once we disable debug log lines

File: bbb/services.py [hunk 2]
```diff
@@ -285,6 +285,16 @@ def reflectTasks(self):
             # If takenUntil isn't set, this task has either never been claimed
             # or got cancelled.
             if not t.takenUntil:
```
Comment (lundjordan) [inline bbb/services.py]: I wonder if we should do, `if not t.takenUntil and not arrow.now() < arrow.get(t.processedDate).replace(minutes=5):`

so then we end up here: https://github.com/mozilla/buildbot-bridge/blob/ed43e28e0e6041691c7bcc4866cd0c788a24df1c/bbb/services.py#L312

 which gives us the same logic flow as what you are doing in this patch without the added complexity.

Comment (rail) [inline bbb/services.py]: I don't have a strong preference here... I'd say in the current implementation you can have separate comments for each case.

Comment (lundjordan) [inline bbb/services.py]: sure, wfm

**Pull request merged**